ci/Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,445 @@
+  sh "kubectl logs ${TEST_KAFKA_POD_NAME} --namespace=${namespace} > ${logFile} || true"
+  archiveArtifacts allowEmptyArchive: true, artifacts: "${logFile}"
+}
+
+void gitCheckout(String reference, String option = '') {
+  sh """
+    git checkout ${option} ${reference}
+  """
+}
+
+void gitPush(String reference) {
+  sh """
+    git push origin ${reference}
+  """
+}
+
+void gitTag(String tagname, String message) {
+  sh """
+    git tag -a ${tagname} -m "${message}"
+  """
+}
+
+pipeline {
+  agent {
+    label 'jenkins-nuxeo-platform-lts-2021'
+  }
+  options {
+    disableConcurrentBuilds()
+    buildDiscarder(logRotator(daysToKeepStr: '15', numToKeepStr: '10', artifactNumToKeepStr: '5'))
+  }
+  environment {
+    DEFAULT_CONTAINER = 'maven'
+    TEST_KAFKA_K8S_OBJECT = 'kafka'
+    TEST_KAFKA_PORT = '9092'
+    TEST_KAFKA_POD_NAME = "${TEST_KAFKA_K8S_OBJECT}-0"
+    TEST_SERVICE_DOMAIN_SUFFIX = 'svc.cluster.local'
+    SLACK_CHANNEL = 'platform-notifs'
+    CONNECT_PREPROD_URL = 'https://nos-preprod-connect.nuxeocloud.com/nuxeo'
+    NUXEO_PACKAGE_PATH = 'nuxeo-compound-documents-package/target/nuxeo-compound-documents-package-*.zip'
+    MAVEN_ARGS = getMavenArgs()
+    VERSION = getVersion()
+    CURRENT_NAMESPACE = getCurrentNamespace()
+    TEST_NAMESPACE_PREFIX = "$CURRENT_NAMESPACE-compound-documents-unit-tests-$BRANCH_NAME-$BUILD_NUMBER".toLowerCase()
+    HELMFILE_COMMAND = "helmfile --file ci/helm/helmfile.yaml --helm-binary /usr/bin/helm3"
+    HOME = '/root'
+  }
+  stages {
+    stage('Set Labels') {
+      steps {
+        container(DEFAULT_CONTAINER) {
+          setLabels()
+        }
+      }
+    }
+    stage('Update Version') {
+      steps {
+        container(DEFAULT_CONTAINER) {
+          script {
+            echo """
+            ----------------------------------------
+            Update version
+            ----------------------------------------
+            New version: ${VERSION}
+            """
+            sh "mvn ${MAVEN_ARGS} versions:set -DnewVersion=${VERSION} -DgenerateBackupPoms=false"
+          }
+        }
+      }
+    }
+    stage('Git commit') {
+      when {
+        allOf {
+          not {
+            branch 'PR-*'
+          }
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
+      }
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Git commit
+          ----------------------------------------
+          """
+          sh """
+            git commit -a -m "Release ${VERSION}"
+          """
+        }
+      }
+    }
+
+    stage('Build') {
+      steps {
+        container(DEFAULT_CONTAINER) {
+          setGitHubBuildStatus('maven/build', 'Build', 'PENDING')
+          sh "mvn ${MAVEN_ARGS} -V -T4C -DskipTests install"
+        }
+      }
+      post {
+        success {
+          archiveArtifacts allowEmptyArchive: true, artifacts: "${NUXEO_PACKAGE_PATH}"
```

Comment:
Let's only archive the package if not a PR.

Reply:
Just checking if I implemented the conditional correctly here

Reply:
I had a suspicion but wanted to see if I could check it without script 😅